db/memtable.h

```diff
@@ -332,6 +333,16 @@ class MemTable {
       num_deletes_.fetch_add(update_counters.num_deletes,
                              std::memory_order_relaxed);
     }
+    if (update_counters.num_range_deletes > 0) {
+      uint64_t val = num_range_deletes_.load(std::memory_order_relaxed) +
+                     update_counters.num_range_deletes;
+      num_range_deletes_.store(val, std::memory_order_relaxed);
+      if (memtable_max_range_deletions_ > 0 &&
+          memtable_max_range_deletions_reached_ == false &&
+          val >= (uint64_t)memtable_max_range_deletions_) {
+        memtable_max_range_deletions_reached_ = true;
```

Comment:
This feels like a data race. What about comparing num_range_deletes_ against memtable_max_range_deletions_ in ShouldFlushNow()?

Reply:
Related (similar scenario where the flag started with one value and could only ever flip once to the other value): #4801

Reply:
Updated to check `num_range_deletes_` against `memtable_max_range_deletions_` in ShouldFlushNow().